hs/src/Reach/Eval/Core.hs

```diff
@@ -3120,9 +3122,27 @@ data CompiledForkCase = CompiledForkCase
   , cfc_req_prop :: JSObjectProperty
   , cfc_switch_case :: JSSwitchParts
   , cfc_only :: JSStatement
+  , cfc_msg_type_def :: [JSStatement]
   }
 
-doFork :: [JSStatement] -> [(SrcLoc, (JSExpression, JSExpression, JSExpression, JSExpression))] -> Maybe (SrcLoc, [JSExpression]) -> App SLStmtRes
+forkCaseAt :: ForkCase -> SrcLoc
```

Comment:
This feels like you really wanted to define a record type with its own accessors